### PR TITLE
[nnfw] Support setInputDimension with nnfw

### PR DESCRIPTION
**Pull request by kparichay**
With updated nnfw API, support setInputDimension with nnfw
Although the API is defined, its not yet implemeted interally
So, the unittest is added, but disabled

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 1]
```diff
@@ -44,6 +41,9 @@
 #define NNFW_SRCN_BACKEND  "srcn"
 #define NNFW_DEFAULT_BACKEND NNFW_CPU_BACKEND
 
+/** Maximum rank allowed for tensor dimension */
+#define NNFW_TENSOR_RANK_LIMIT 6
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Oh, does NNFW allow tensor rank up to 6 ??

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: [Yes](https://review.tizen.org/gerrit/gitweb?p=platform/core/ml/nnfw.git;a=blob;f=runtime/neurun/api/include/nnfw.h;h=1e18deaab332fb2c05a4a5e3810de3ce6825ab38;hb=refs/heads/tizen#l117)

Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Then, shouldn't it be defined in a header file of nnfw?

Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Anyway, https://lists.fedoraproject.org/pipermail/scm-commits/Week-of-Mon-20150413/1558560.html says "4" is the limit...

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 2]
```diff
@@ -175,10 +177,17 @@ nnfw_open (const GstTensorFilterProperties * prop, void **private_data)
...
   }
 
-  if (nnfw_tensors_info_get (pdata, TRUE, &pdata->in_info) ||
-      nnfw_tensors_info_get (pdata, FALSE, &pdata->out_info)) {
-    err = -EINVAL;
-    g_printerr ("Error retrieving input/output info from nnfw-runtime.");
+  err = nnfw_tensors_info_get (pdata, TRUE, &pdata->in_info);
+  if (err) {
+    g_printerr ("nnfw-runtime cannot prepare the session for %s",
+        prop->model_files[0]);
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Mistakes in error messages?

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Updated to be more specific.

**Comment by myungjoo** [comment]
You may refer https://github.sec.samsung.net/STAR/nnfw/pull/9857 later.

**Comment by kparichay** [comment]
> You may refer #9857 of nnfw later.

Although this is merged, I will wait for this to reflect in review.tizen.org.

**Pull request merged**